### DISCUSSION TITLE
How to make window configuration persistent between refreshing?

### DISCUSSION THREAD

**Discussion by mcnuggetsx20** [qtile/qtile#4136] [Q&A]
Let's say I have two windows open. When I shuffle them and then refresh qtile they come back to their original positions. How do I change that?

**Comment by jrwrigh** [comment]
I'm not aware of a way to do that, but would be happy if someone corrected me on that!

**Comment by m-col** [comment]
You can't do that unfortunately (currently). Both restarting Qtile and reloading the config persist state across the restart/reload in this class: https://github.com/qtile/qtile/blob/master/libqtile/core/state.py. 

Before restart/reload, it remembers the groups, screens, and the current screen, and then the state object gets pickled while Qtile does other stuff.
After restart/reload, the state object gets unpickled and it attempts to match current groups and screens with those that were remembered.

You want layout state to also get persisted. For this, each layout would need a way to provide and accept state that is sufficient for its state to appear unchanged across a restart. I'm not sure how simple a task would be, and a default implementation on the base layout that does nothing would probably be fine. But PR welcome.
